[ventaair] New VentaAir binding for air humidifiers (#9979)
openhab/openhab-addons: bundles/org.openhab.binding.ventaair/README.md

Thankfully the vendor allows for communicating within the local network without needing any internet access or accounts.
This is even stated in the official manual.
Hence this binding communicates locally with the humidifier and is able to read out the current measurements and settings, as well as changing the settings.

## Supported Things

It currently supports the LW60-T device as well as a generic `ThingType` for other models.
For now the generic `ThingType` only adds the "boost" channel, but in the status reply from the device there is more which could be added in the future by someone who owns a different device.

## Discovery

This binding supports an automatic discovery for humidifiers that are connected to the local network and which are in the same broadcast domain.
To do so, the binding listens to UDP port 48000 for data and creates `DiscoveryResult`s based on the received data from the device.
This comes in handy for getting the MAC address for the device for example.
Once the `DiscoveryResult` is added as a `Thing`, a connection to the device will be created and it will beep, showing a confirmation screen that the device "openHAB" would like to get access.

> After confirming this request, the use can link its `Item`s to receive data or control the device.

Reviewer: ```suggestion
After confirming this request, the user can link items to receive data or control the device.
```

## Thing Configuration

There are three mandatory configuration parameters for a thing: `ipAddress`, `macAddress` and `deviceType`.

> Whereas the first two are self-exploring, the later one defines the type of the device. It is an integer value and its best to use the automatic discovery to obtain it from the device.

Reviewer: ```suggestion
Whereas the first two are self-explaining, the later one defines the type of the device.
It is an integer value and its best to use the automatic discovery to obtain it from the device.
```

How many device types are there? Have you considered making multiple thing types instead?

Author: Yes, of course I have thought about it. However, I do a) not own any other device and b) I thought a more general solution to catch ALL devices would be best for the normal user.

That's why there are just the 2 thing types. One for the device that I own and a generic one.

The optional parameter `pollingTime` is the time interval in seconds in which the data should be polled from the device, default is 10 seconds.

> The `hash` parameter is also optional. It is a negative integer value and it is used by the device to identify a connection to a client, like the App from the vendor for example.

Reviewer: It may be better to have this information in a table format, check the other bindings on how they do this. You have done it for the channels already which is good. Also each sentence should be on its own line.

> I do not know whether there devices that are restricted to only one client, so I added this parameter to allow the user to set the same value as his App on the phone (can be obtained via sniffing the network).

Reviewer: ```suggestion
I do not know whether their devices are restricted to only one client, so I added this parameter to allow the user to set the same value as the app on their phone (can be obtained via sniffing the network).
```

## Supported Things

It currently supports the LW60-T device as well as a generic `ThingType` for other models.
For now the generic `ThingType` only adds the "boost" channel, but in the status reply from the device there is more which could be added in the future by someone who owns a different device.

## Discovery

This binding supports an automatic discovery for humidifiers that are connected to the local network and which are in the same broadcast domain.
To do so, the binding listens to UDP port 48000 for data and creates `DiscoveryResult`s based on the received data from the device.
This comes in handy for getting the MAC address for the device for example.
Once the `DiscoveryResult` is added as a `Thing`, a connection to the device will be created and it will beep, showing a confirmation screen that the device "openHAB" would like to get access.
After confirming this request, the user can link its items to receive data or control the device.

## Thing Configuration

There are three mandatory configuration parameters for a thing: `ipAddress`, `macAddress` and `deviceType`.

> | parameter  | mandatory/optional description |

Reviewer: ```suggestion
| parameter  | mandatory/optional | description |
```

You could also use "required" as the heading and a Y and N, up to you.

| hash | N | It is a negative integer value and it is used by the device to identify a connection to a client, like the App from the vendor for example. (*) |

(*) I do not know whether there are devices which are restricted to only one client, so I added this parameter to allow the user to set the same value as his App on the phone (can be obtained via sniffing the network).
However, the LW60-T allows for multiple connections to different clients, identified by different `hash` values at the same time without issues.
By default the binding uses "-42", so a new ID that is not known to the device and hence it asks for confirmation, see the Discovery section.

Example Thing configuration:

```
Thing ventaair:lw60t:humidifier [ ipAddress="192.168.42.69", macAddress="f8:f0:05:a6:4e:03", deviceType=4, pollingTime=10, hash=-42]
```

## Channels

These are the channels that are currently supported:

| channel  | type (RO=read-only)   | description                  |
|----------|--------|------------------------------|
| power          | Switch | This is the power on/off channel  |

> | fanSpeed       | Number | This is the channel to control the steps for the speed of the fan |

Reviewer 2: You could mention the range or unit of the Channel.

> | targetHumidity | Number | This channel sets the target humidity that should be try to reach by the device  |

Reviewer 2: You could mention the range and allowed values.

> | timer          | Number | This channel sets the power off timer to the set value in hours, i.e. 3 = turn off in 3 hours from now |

Reviewer 2: You could mention the range and allowed values.

| sleepMode      | Switch | This channel controls the sleep mode of the device (dims the display and slows down the fan) |
| childLock      | Switch | This is the control channel for the child lock |
| automatic      | Switch | This is the control channel to start the automatic operation mode of the device |
| cleanMode      | Switch (RO) | This is the channel that indicates if the device is in the cleaning mode |
| temperature    | Number:Temperature (RO) | This channel provides the current measured temperature in Celsius or Fahrenheit as configured on the device |

> | humidity       | Number (RO) | This channel provides the humidity measured by the device in percent |

Reviewer 2: You could use `QuantityType` here with `Units.PERCENT`.
```suggestion
| humidity       | Number:Dimenionsless (RO) | This channel provides the humidity measured by the device in percent |
```

> | humidity       | Number:Dimenionsless (RO) | This channel provides the humidity measured by the device in percent |

Reviewer 2: ```suggestion
| humidity       | Number:Dimensionless (RO) | This channel provides the humidity measured by the device in percent |
```

| waterLevel     | Number (RO) | This channel indicates the water level of the tank where 1 is equal to the yellow "refill tank" warning on the device/App |
| fanRPM         | Number (RO) | This channel provides the speed of the ventilation fan |

> | timerTimePassed  | Number (RO) | If a timer has been set, this channel provides the minutes since when the timer was started  |

Reviewer 2: Same for below.
```suggestion
| timerTimePassed  | Number:Time (RO) | If a timer has been set, this channel provides the minutes since when the timer was started  |
```
